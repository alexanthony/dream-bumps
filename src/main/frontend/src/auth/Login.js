import React, {PropTypes} from 'react';
import {connect} from 'react-redux';
import {Link} from 'react-router';
// import {Col} from 'react-flexbox-grid';
import { Cell } from 'material-grid/dist';
import {Field, reduxForm} from 'redux-form';
import {TextField} from 'redux-form-material-ui';
import RaisedButton from 'material-ui/RaisedButton';
import {Card, CardTitle, CardText, CardActions} from 'material-ui/Card';

import {login} from './actions';

export const Login = ({handleSubmit}) => (
  <form onSubmit={handleSubmit}>
    <Cell col={6} offset={3}>
      <Card>
        <CardTitle title="Login"/>
        <CardText>
          <div>
            Not got an account yet? <Link to="/register">Register</Link>
          </div>
          <Field name="username" component={TextField} floatingLabelText="Username" fullWidth/>
          <Field name="password" component={TextField} floatingLabelText="Password" type="password" fullWidth/>
        </CardText>
        <CardActions>
          <RaisedButton primary label="Log in" type="submit"/>
        </CardActions>
      </Card>
    </Cell>
  </form>
);

Login.propTypes = {
  handleSubmit: PropTypes.func
};

const LoginForm = reduxForm({
  form: 'login',
  fields: [
    'username', 'password', 'rememberMe'
  ],
  initialValues: {
    rememberMe: true
  }
})(Login);

export default connect(null, {onSubmit: login})(LoginForm);
