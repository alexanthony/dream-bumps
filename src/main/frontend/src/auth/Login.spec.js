import React from 'react';
import renderer from 'react-test-renderer';
import { Login } from './Login';

test('Renders properly', () => {
  const component = renderer.create(
    <Login />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

});
