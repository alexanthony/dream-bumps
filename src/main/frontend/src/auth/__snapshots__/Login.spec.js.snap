exports[`test Renders properly 1`] = `
<form>
  <Cell
    desktop={6}
    desktopOffset={3}
    tablet={6}
    tabletOffset={1}>
    <Card
      expandable={false}
      expanded={null}
      initiallyExpanded={false}>
      <CardTitle
        title="Login" />
      <CardText>
        <div>
          Not got an account yet? 
          <Link
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/register">
            Register
          </Link>
        </div>
        <Field
          component={[Function]}
          floatingLabelText="Username"
          fullWidth={true}
          name="username" />
        <Field
          component={[Function]}
          floatingLabelText="Password"
          fullWidth={true}
          name="password"
          type="password" />
      </CardText>
      <CardActions>
        <RaisedButton
          disabled={false}
          fullWidth={false}
          label="Log in"
          labelPosition="after"
          primary={true}
          secondary={false}
          type="submit" />
      </CardActions>
    </Card>
  </Cell>
</form>
`;
